pkg/apis/configuration/v1alpha1/kongcustomentity_types.go

```diff
@@ -0,0 +1,144 @@
+package v1alpha1
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kongv1 "github.com/kong/kubernetes-ingress-controller/v2/pkg/apis/configuration/v1"
+)
+
+// +kubebuilder:object:root=true
+
+// KongCustomEntityList is a list of kong custom entities.
+type KongCustomEntityList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []KongCustomEntity `json:"items"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=kongcustomentities,shortName=kce,categories=kong-ingress-controller
+// +kubebuilder:subresource:status
+
+// KongCustomEntity represents a custom entity in Kong.
+type KongCustomEntity struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec is the specification of the entity.
+	Spec KongCustomEntitySpec `json:"spec"`
+	// Status is the status of the entity.
+	Status KongCustomEntityStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:
```

Comment:
Are those `// +k8s:deepcopy-gen:` really required? Having just `// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object` (line 20) is not enough? In other CRDs defined by Kong we don't have those dangling `// +k8s:deepcopy-gen:`. 🤔 

The same for any other appearance of it